src/test/java/hudson/plugins/active_directory/ActiveDirectorySecurityRealmFipsAndForceLdapEnabledTest.java

```diff
@@ -0,0 +1,64 @@
+package hudson.plugins.active_directory;
+
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.recipes.LocalData;
+
+import hudson.util.FormValidation;
+
+import static org.junit.Assert.assertEquals;
+
+import static hudson.plugins.active_directory.ActiveDirectorySecurityRealm.LEGACY_FORCE_LDAPS_PROPERTY;
+
+/*
+Verifies the warning message when Jenkins operates in FIPS and Force LDAP property is enabled.
+ */
+public class ActiveDirectorySecurityRealmFipsAndForceLdapEnabledTest {
+
+    @Rule
+    public JenkinsRule jenkinsRule = new JenkinsRule();
+
+    @BeforeClass
+    public static void globalSetup() {
+        System.setProperty("jenkins.security.FIPS140.COMPLIANCE", "TRUE");
+        System.setProperty(LEGACY_FORCE_LDAPS_PROPERTY, "TRUE");
+    }
+
+    @LocalData
+    @Test
+    public void testStartTlsForWarnings() throws Exception {
+        ActiveDirectorySecurityRealm activeDirectorySecurityRealm = (ActiveDirectorySecurityRealm) jenkinsRule.jenkins.getSecurityRealm();
+
+        FormValidation result = activeDirectorySecurityRealm.getDescriptor().doCheckStartTls(false, false);
+        assertEquals(FormValidation.Kind.OK, result.kind);
```

Comment:
In FIPS mode, if the two of them are false you shouldn't get an OK ... unless the System property is set. My first intention was to say this was wrong, that's why it's recommendable always to add a String with an explanation
```suggestion
        assertEquals("FIPS mode and legacy system property configured, so it's compliant", FormValidation.Kind.OK, result.kind);
```